Source/DafnyCore/Pipeline/Compilation.cs

```diff
@@ -518,6 +519,9 @@ CompilationInput input
     foreach (var counterExample in result.CounterExamples) //.OrderBy(d => d.GetLocation()))
     {
       var errorInformation = counterExample.CreateErrorInformation(outcome, options.ForceBplErrors);
+      if (options.ShowProofObligationExpressions && errorInformation.Kind != null) {
```

Comment:
The compiler is warning that the `errorInformation.Kind != null` condition is guaranteed to be true and is unnecessary.